crates/sui-rpc-loadgen/src/payload/rpc_command_processor.rs

```diff
@@ -211,14 +212,168 @@ impl<'a> ProcessPayload<'a, &'a GetCheckpoints> for RpcCommandProcessor {
+                                }
+                                check_transactions(&self.clients, &t.transactions).await;
+                                Some(t)
+                            },
+                            Err(err) => {
+                                error!("Failed to fetch checkpoint {seq} on the {i}th url: {err}");
+                                None
+                            }
+                        };
+                        let elapsed_time = start_time.elapsed();
+
+                        println!(
+                            "GetCheckpoint Request latency {:.4}",
+                            elapsed_time.as_secs_f64(),
+                        );
+                        checkpoint
+                    }
+                }
+            }))
+            .await;
+
+                // TODO(chris): read `cross_validate` from config
+                let cross_validate = true;
+                if cross_validate {
+                    let (valid_checkpoint_idx, valid_checkpoint) = checkpoints
+                        .iter()
+                        .enumerate()
+                        .find_map(|(i, x)| {
+                            if x.is_some() {
+                                Some((i, x.clone().unwrap()))
+                            } else {
+                                None
+                            }
+                        })
+                        .expect("There should be at least one RPC server returning a checkpoint");
+                    for (i, x) in checkpoints.iter().enumerate() {
+                        if i == valid_checkpoint_idx {
+                            continue;
+                        }
+                        // ignore the None value because it's warned above
+                        let eq = x.is_none() || x.as_ref().unwrap() == &valid_checkpoint;
+                        if !eq {
+                            error!("getCheckpoint {seq} has a different result between the {valid_checkpoint_idx}th and {i}th URL {:?} {:?}", x, checkpoints[valid_checkpoint_idx])
+                        }
+                    }
+                }
+
+                if seq % 100 == 0 {
+                    debug!("Finished processing checkpoint {seq}")
+                }
+            }
+        }
         Ok(())
     }
 }
 
+pub async fn check_transactions(
+    clients: &Arc<RwLock<Vec<SuiClient>>>,
+    digests: &[TransactionDigest],
+) {
+    let read = clients.read().await;
+    let clients = read.clone();
+
+    let transactions = join_all(clients.iter().enumerate().map(|(i, client)| async move {
+        let start_time = Instant::now();
+        let transactions = client
+            .read_api()
+            .multi_get_transactions_with_options(
+                digests.to_vec(),
+                SuiTransactionResponseOptions::new(),
```

Comment:
Ah, this by default is only going to return `digest`. You need to specify some options, let's start with input and effects